tests/cli.rs

```diff
@@ -0,0 +1,236 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
here are a bunch of tests for the CLI

Reply:
We can probably move some of these to a `common` or something similar test since it covers both the CLI and TUI

Reply:
I think that makes sense.  we'll need to do some testing of the execution triggered by the UI but it definitely doesnt need to be concerned with details - more just making sure the results make it into app state.